<div style='background-color: orange'>
<a id='TableOfContents'></a>
    <b><u><i><h1 style='text-align: center'>
        Table of Contents
    </h1></i></u></b>
<li><a href='#imports'>Imports</a></li>
<li><a href='#wrangle'>Wrangle</a></li>
<li><a href='#explore'>Exploration</a></li>
<li><a href='#model'>Modeling</a></li>
<li><a href='#conclusion'>Conclusion</a></li>

<div style='background-color: orange'>
<a id='imports'></a>
    <b><u><i><h1 style='text-align: center'>
        Imports
    </h1></i></u></b>
<li><a href='#TableOfContents'>Table of Contents</a></li>

In [1]:
# Import 'final.py' to expedite and maintain cleanliness
# of this file...
import final as f

<div style='background-color: orange'>
<a id='wrangle'></a>
    <b><u><i><h1 style='text-align: center'>
        Wrangle
    </h1></i></u></b>
<li><a href='#TableOfContents'>Table of Contents</a></li>

<h3><b>Acquire</b></h3>

Obtain 'mass_shooters' dataset from excel sheet given by:

<a href='https://www.theviolenceproject.org/'>Non-Profit Organization: 'The Violence Project'</a>

In [2]:
# Using the 'final.py' file...
# Acquire the vanilla 'mass_shooters' dataframe...
vanilla_mass_shooters = f.acquire()
vanilla_mass_shooters.shape

(189, 153)

- Vanilla 'mass_shooters' shape
    - Rows: 189
    - Columns: 153

---

<h3><b>Prepare</b></h3>

In [3]:
# From the 'final.py' file...
# Get the prepared 'mass_shooters' dataset
prepped_mass_shooters = f.prepare()
prepped_mass_shooters.shape

(188, 254)

<a id='preparesummary'></a>
<h3><b><i>
    Summary of Preparation
</i></b></h3>
<li><a href='#prepare'>Prepare Top</a></li>

- <h5><b>Dropped Data</b></h5>

    - 1 Row
        - Missing majority of info
    - 47 columns
        - Useless in scope of predictive value
        - Percent nulls above 20%
        
- <h5><b>Null Handling</b></h5>

    - 44 columns filled with mode
    - 1 column filled with 'None'

- <h5><b>Dtype Cleaning</b></h5>
    
    - Object
        - 4 columns to int
        - 3 columns fixed
    - Float
        - 79 columns to int
        - 3 columns fixed
    - Int
        - Nothing changed
    
- <h5><b>Disseminate Column Information</b></h5>
    
    - 31 columns manually disseminated
        - 137 new columns generated
    
- <h5><b>Aggregate Column Creation</b></h5>    

    - Columns Created: 13

- <h5><b>Text Modifications</b></h5>

    - Column Name Changes
        - Removals: 5
            - '\xa0'
            - '('
            - ')'
            - '/'
            - ':'
        - Replacements: 2
            - whitespace ==> '_'
            - '#' ==> 'id'
        - Case Change: 1
            - ALL LOWERCASE

- <h5><b>Target Column Creation</b></h5>

    - Target value is predicting the mass shooter's volatility as accurately as possible based on the number of casualties
        - 'High Volatility'
            - Casualties > 10
        - 'Low Volatility'
            - Casualties <= 10

- <h5><b>Final Dataframe</b></h5>

    - Rows: 188
    - Columns: 254

---

<h3><b>Split</b></h3>

In [4]:
# From the 'final.py' file...
# Split the prepared 'mass_shooters' dataframe
train, validate, test = f.wrangle()
train.shape, validate.shape, test.shape

((131, 254), (38, 254), (19, 254))

- Train
    - 70%
- Validate
    - 20%
- Test
    - 10%

<div style='background-color: orange'>
<a id='explore'></a>
    <b><u><i><h1 style='text-align: center'>
        Exploration
    </h1></i></u></b>
<li><a href='#TableOfContents'>Table of Contents</a></li>
<li><a href='#visuals'>Key Visuals</a></li>
<li><a href='#exploresummary'>Summary</a></li>

<a id='visuals'></a>
<h3><b>Key Visuals</b></h3>
<li><a href='#explore'>Exploration Top</a></li>

<b>Question 1:</b> 

<b>H$_0$:</b> 

<b>H$_a$:</b> 

In [5]:
# From the 'final.py' file...
# Get the histogram of {FEATURE} vs. {FEATURE}


In [6]:
# From the 'final.py' file...
# Get the {STAT} of {FEATURE} vs. {FEATURE}


<b>Q1 Answer:</b> 

---

<b>Question 2:</b> 

<b>H$_0$:</b> 

<b>H$_a$:</b> 

In [7]:
# From the 'final.py' file...
# Get the {PLOT} of {FEATURE} vs. {FEATURE}


In [8]:
# From the 'final.py' file...
# Get the {STAT} of {FEATURE} vs. {FEATURE}


<b>Q2 Answer:</b> 

---

<b>Question 3:</b> 

<b>H$_0$:</b> 

<b>H$_a$:</b> 

In [9]:
# From the 'final.py' file...
# Get the {PLOT} of {FEATURE} vs. {FEATURE}


In [10]:
# From the 'final.py' file...
# Get the {STAT} of {FEATURE} vs. {FEATURE}


<b>Q3 Answer:</b> 

---

<b>Question 4:</b> 

<b>H$_0$:</b> 

<b>H$_a$:</b> 

In [11]:
# From the 'final.py' file...
# Get the {PLOT} of {FEATURE} vs. {FEATURE}


In [12]:
# From the 'final.py' file...
# Get the {STAT} of {FEATURE} vs. {FEATURE}


<b>Q4 Answer:</b> 

---

<a id='exploresummary'></a>
<h3><b>Summary</b></h3>
<li><a href='#explore'>Exploration Top</a></li>

SUMMARY OF EXPLORATION

<div style='background-color: orange'>
<a id='model'></a>
    <b><u><i><h1 style='text-align: center'>
        Modeling
    </h1></i></u></b>
<li><a href='#TableOfContents'>Table of Contents</a></li>
<li><a href='#baseline'>Baseline</a></li>
<li><a href='#modeltable'>Models Table</a></li>
<li><a href='#bestmodel'>Best Model</a></li>

<a id='baseline'></a>
<h3><b>Baseline</b></h3>
<li><a href='#model'>Modeling Top</a></li>

In [13]:
# From the 'final.py' file...
# Obtain the baseline score of {FEATURE} via {METHOD}


---

<a id='modeltable'></a>
<h3><b>Models Table</b></h3>
<li><a href='#model'>Modeling Top</a></li>

In [14]:
# From the 'final.py' file...
# Get the scores of the baseline and 4 unique models


{METRIC HERE}:

{EXPLANATION OF REASONING HERE}

---

<a id='bestmodel'></a>
<h3><b>Best Model</b></h3>
<li><a href='#model'>Modeling Top</a></li>

In [15]:
# From the 'final.py' file...
# Get the scores of the baseline and the best model


<div style='background-color: orange'>
<a id='conclusion'></a>
    <b><u><i><h1 style='text-align: center'>
        Conclusion
    </h1></i></u></b>
<li><a href='#TableOfContents'>Table of Contents</a></li>
<li><a href='#conclusionsummary'>Summary</a></li>
<li><a href='#rec'>Recommendations</a></li>
<li><a href='#nextsteps'>Next-Steps</a></li>

<a id='conclusionsummary'></a>
<h3><b>Summary</b></h3>
<li><a href='#conclusion'>Conclusion Top</a></li>

SUMMARY HERE

---

<a id='rec'></a>
<h3><b>Recommendations</b></h3>
<li><a href='#conclusion'>Conclusion Top</a></li>

RECOMMENDATIONS HERE

---

<a id='nextsteps'></a>
<h3><b>Next-Steps</b></h3>
<li><a href='#conclusion'>Conclusion Top</a></li>

NEXT-STEPS HERE